sherpa/astro/ui/utils.py

```diff
@@ -10431,13 +10431,78 @@ def get_fit_plot(self, id=None, recalc=True):
 
     get_fit_plot.__doc__ = sherpa.ui.utils.Session.get_fit_plot.__doc__
 
-    # copy doc string from sherpa.utils
     def get_model_component_plot(self, id, model=None, recalc=True):
+        """Return the data used to create the model-component plot.
+
+        .. versionchanged :: 4.13.2
+           For PHA data, the response model is now automatically added
+           by the routine unless the model contains a response.
+
+        Parameters
+        ----------
+        id : int or str, optional
+           The data set that provides the data. If not given then the
+           default identifier is used, as returned by `get_default_id`.
+        model : str or sherpa.models.model.Model instance
+           The component to use (the name, if a string).
+        recalc : bool, optional
+           If ``False`` then the results from the last call to
+           `plot_model_component` (or `get_model_component_plot`)
+           are returned, otherwise the data is re-generated.
+
+        Returns
+        -------
+        instance
+           An object representing the data used to create the plot by
+           `plot_model_component`. The return value depends on the
+           data set (e.g. PHA, 1D binned, or 1D un-binned).
+
+        See Also
+        --------
+        get_model_plot : Return the data used to create the model plot.
+        plot_model : Plot the model for a data set.
+        plot_model_component : Plot a component of the model for a data set.
+
+        Notes
+        -----
+        The function does not follow the normal Python standards for
+        parameter use, since it is designed for easy interactive use.
+        When called with a single un-named argument, it is taken to be
+        the `model` parameter. If given two un-named arguments, then
+        they are interpreted as the `id` and `model` parameters,
+        respectively.
+
+        Examples
+        --------
+
+        Return the plot data for the ``pl`` component used in the
+        default data set:
+
+        >>> cplot = get_model_component_plot(pl)
+
+        Return the full source model (``fplot``) and then for the
+        components ``gal * pl`` and ``gal * gline``, for the data set
+        'jet':
+
+        >>> fmodel = xsphabs.gal * (powlaw1d.pl + gauss1d.gline)
+        >>> set_source('jet', fmodel)
+        >>> fit('jet')
+        >>> fplot = get_model('jet')
+        >>> plot1 = get_model_component_plot('jet', pl*gal)
```

Comment:
This sounds as if `get_model` and `get_model_component_plot` are essentially identical. Yes, one of them is names `plot` and the other one is not. As a reader, I wonder what the difference is? While one get me a model object and the other one a figure object?

Reply:
This is likely an error in the example code - it probably (will have to check) should have said

    >>> fplot = get_model_plot('jet')

Reply:
I've made this change (replacing get_model with get_model_plot)